.github/workflows/ci.yaml

```diff
@@ -34,14 +34,14 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
         pydantic-version: ["1", "2"]
     uses: ./.github/workflows/test.yaml
     with:
-      coverage: ${{ matrix.python-version == '3.11' && matrix.pydantic-version == '2' }}
+      coverage: ${{ matrix.python-version == '3.12' && matrix.pydantic-version == '2' }}
```

Comment:
I recommend not to use coverage with 3.12 just yet, there are some possible (and known) bugs, which can result in invalid coverage percentage or line numbers. 3.11 is fine.